### PR TITLE
Fix some clippy warnings

### PR DESCRIPTION
**Pull request by mariocao**
Just an update of the tokio library to the last version, and some minor fixes for some warnings identified by `cargo clippy`.

### CODE REVIEW

**Comment by DoumanAsh** [comment]
Current actix should use latest tokio already, you're more or less bumping minimum version.

Is it necessary to stop using separate tokio crates and use only meta crate? I don't think there is a need to remove it as we might find it useful later on

**Comment by mariocao** [comment]
You are right. As tokio project moved its crates as sub crates, I thought it could be a good practice to follow less number of versions into a single one. So, if you think it is better like that, close the PR. :smile:

**Comment by DoumanAsh** [comment]
Fixing warnings is still good idea though 👍

**Comment by mariocao** [comment]
I removed the tokio stuff and left only the fixes of clippy warnings :)
I will edit the PR title in order not to be misleading.

File: src/registry.rs [hunk 1]
```diff
@@ -290,10 +290,10 @@ impl SystemRegistry {
     }
 
     /// Add new actor to the registry by address, panic if actor is already running
-    pub fn set<A: SystemService + Actor<Context = Context<A>>>(&self, addr: Addr<A>) {
+    pub fn set<A: SystemService + Actor<Context = Context<A>>>(&self, addr: &Addr<A>) {
```
Comment (DoumanAsh) [inline src/registry.rs]: hm... This is technically breaking change as it is public interface.

I think instead we should not clone addr below at line 302.
The effect would be the same

Comment (mariocao) [reply src/registry.rs]: :+1:  :)

**Pull request closed without merging**